frontend/src/views/admin/matching/index.tsx

```diff
@@ -1,14 +1,153 @@
 import React from "react";
+
 import { ContentArea } from "../../../components/layout";
 import { useSelector } from "react-redux";
 import { activeSessionSelector, fetchPostings } from "../../../api/actions";
 import { useThunkDispatch } from "../../../libs/thunk-dispatch";
-import { GroupedOverview } from "./grouped-overview";
+import { round } from "../../../libs/utils";
+
+import {
+    applicationsSelector,
+    assignmentsSelector,
+    positionsSelector,
+    applicantsSelector,
+} from "../../../api/actions";
+import { Assignment, Application, Applicant } from "../../../api/defs/types";
+import { MatchingDataState } from "./reducers";
+
+import {
+    matchingDataSelector,
+    matchesSelector,
+    guaranteesSelector,
+    notesSelector,
+    batchUpsertMatches,
+    batchUpsertGuarantees,
+    batchUpsertNotes,
+} from "./actions";
+import {
+    PositionSummary,
+    ApplicantSummary,
+    Match,
+    AppointmentGuaranteeStatus,
+} from "./types";
+
+import { PositionList } from "./position-list";
+import { ApplicantView } from "./applicant-view";
+
+import {
+    ImportMatchingDataButton,
+    ImportGuaranteesButton,
+    ExportMatchingDataButton,
+} from "./import-export";
+import { FinalizeChangesButton } from "./finalize-changes";
+import { Button } from "react-bootstrap";
+
+function getCombinedApplication(
```

Comment:
You're going to call this over and over again, right? You should probably make a selector out of this and just have it compute everyone's combined application efficiently. Then you can select the relevant combined application.

If you use `createSelector` (or whatever it's called) to do this, then the library will memoize it automatically.